lib/ssl/src/tls_connection.erl

```diff
@@ -622,9 +626,15 @@ init({call, From}, {start, Timeout},
                          connection_env = CEnv#connection_env{
                                             negotiated_version = RequestedVersion},
                          session = Session,
-                         handshake_env = HsEnv#handshake_env{tls_handshake_history = Handshake},
+                         handshake_env = HsEnv#handshake_env{
+                             tls_handshake_history = Handshake,
+                             ocsp_stapling_state =
+                                 #{is_server_hello_requested => undefined,
```

Comment:
`is_server_hello_requested` and `is_client_hello_requested` could be replaced by one variable. We have only one shared state for OCSP stapling. First, we evaluate the client options, and then we set a variable here in the state that represent the negotiated state of the stapling function. Before the negotiation it will be set to the same value as the `ocsp_stapling` option. Later we will change its value based on the the received ServerHello extensions (the negotiated value).

Reply:
> `is_server_hello_requested` and `is_client_hello_requested` could be replaced by one variable. We have only one shared state for OCSP stapling. First, we evaluate the client options, and then we set a variable here in the state that represent the negotiated state of the stapling function. Before the negotiation it will be set to the same value as the `ocsp_stapling` option. Later we will change its value based on the the received ServerHello extensions (the negotiated value).

done